src/queueFsm.js

```diff
@@ -74,13 +74,17 @@ var Factory = function( options, connection, topology, serializers, queueFn ) {
       }.bind( this );
 
       var subscriber = function( exclusive ) {
-        queue.subscribe( !!exclusive )
-          .then(
-            onSubscribe,
-            function( err ) {
-              emit( "subscribeFailed", err )
-            }
-          );
+		return when.promise( function( resolve, reject ) {
+			queue.subscribe( !!exclusive )
+			  .then(
+				onSubscribe,
+				function( err ) {
+				  emit( "subscribeFailed", err )
+				  reject("subscribeFailed " + err);
+				}
+			  )
+			  .then(resolve);
+		});
```

Comment:
hi there,

no need to wrap this up in a `return when.promise( ... )` because `queue.subscribe` is already returning a promise, also `subscriber` does not reference `this` so I don't think there is a need to call `bind(this)` the following is (should be) functionally equivalent:

```js
var subscriber = function( exclusive ) {
  return queue.subscribe( !!exclusive ).then(onSubscribe).catch(function(err) { 
    emit( "subscribeFailed", err );
    throw err;
  });
};
```